app/models/services/tumblr.rb

```diff
@@ -39,8 +39,23 @@ def tumblr_template(post, url)
     post.photos.each do |photo|
       html << "![photo](#{photo.url(:scaled_full)})\n\n"
     end
+    # save all the mentions within a post
+    @mentions = []
+    post.message.instance_values["options"][:mentioned_people].each do |o|
+      @person_url = o.url + "people/" + o.guid
+      @mentions << {name: o.name, url: @person_url}
+    end
+    # removes the hovercard link that gets added by default
     html << post.message.html(mentioned_people: [])
     html << "\n\n[original post](#{url})"
+    @mentions.each { |mention|
+      html.gsub!("#{mention[:name]}", "[#{mention[:name]}](#{mention[:url]})")
```

Comment:
Mmmh, this will also replace occurrences of the name anywhere that aren't meant as  mentions. Say I mention a user named "commit" and post a link like https://github.com/foo/bar/commit/132345abdef ...

I'm afraid this likewise needs to add support to Mentionable for generating full URL links.